apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Query/ImbalancePriceQuery.cs

```diff
@@ -78,16 +78,55 @@ static async Task<ImbalancePricePeriod> GetPricesAsync(DateTimeOffset from, Date
     public async Task<IEnumerable<ImbalancePricesDailyDto>> GetImbalancePricesForMonthAsync(
         int year,
         int month,
-        Energinet.DataHub.WebApi.GraphQL.Enums.PriceAreaCode areaCode,
+        Enums.PriceAreaCode areaCode,
         [Service] IImbalancePricesClient_V1 client)
     {
         var parsedAreaCode = areaCode switch
         {
-            Energinet.DataHub.WebApi.GraphQL.Enums.PriceAreaCode.Dk1 => PriceAreaCode.AreaCode1,
-            Energinet.DataHub.WebApi.GraphQL.Enums.PriceAreaCode.Dk2 => PriceAreaCode.AreaCode2,
+            Enums.PriceAreaCode.Dk1 => PriceAreaCode.AreaCode1,
+            Enums.PriceAreaCode.Dk2 => PriceAreaCode.AreaCode2,
             _ => throw new ArgumentOutOfRangeException(nameof(areaCode)),
         };
 
-        return await client.GetByMonthAsync(year, month, parsedAreaCode);
+        var imbalancePrices = await client.GetByMonthAsync(year, month, parsedAreaCode);
+
+        foreach (var imbalancePrice in imbalancePrices)
+        {
+            var missingTimestamps = new List<ImbalancePriceDto>();
+            var sortedPrices = imbalancePrice.ImbalancePrices.OrderBy(x => x.Timestamp);
+
+            var previousTimestamp = sortedPrices.First().Timestamp;
+
+            foreach (var price in sortedPrices.Skip(1))
+            {
+                var currentTimestamp = price.Timestamp;
+
+                var differenceInHours = (currentTimestamp - previousTimestamp).TotalHours;
+
+                if (differenceInHours > 1)
+                {
+                    for (int i = 1; i < differenceInHours; i++)
+                    {
+                        var missingTimestamp = previousTimestamp.AddHours(i);
+                        var missingPrice = new ImbalancePriceDto
+                        {
+                            Timestamp = missingTimestamp,
+                            Price = 0,
+                        };
+
+                        missingTimestamps.Add(missingPrice);
+                    }
+                }
+
+                previousTimestamp = currentTimestamp;
+            }
```

Comment:
issue: the logic should also handle 15 min gaps. There could be days where the prices are in 15 min resolution, e.g.

From T-001

![image](https://github.com/user-attachments/assets/239cb727-cfb2-4a24-8a69-3c313b093bd2)

Reply:
@FirestarJes what gives? :-D what about the cut off date we talked about? :D

Reply:
I was under the impression we had a cut-off date where this was to implemented, wasn't aware it was already implemented. Well if we already have these we need to handle them

Reply:
We are not yet switched to quaterly values, so have overridden it with hourly instead, so the data is right.